be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/handlers/SocialMediaHandler.scala

```diff
@@ -30,14 +36,57 @@ case object SocialMediaHandler extends MessageHandler {
   }
   
   def handleAddChirp(rpcMessage: JsonRpcRequest): GraphMessage = {
-    //FIXME: broadcast chirp On general channel?
-    //create message (Broadcast one)
-    //we can use PubSubMediator.Propagate(channel (fixed), message (created here))
-    // mediatorRef ! PubSubMediator.Propagate(channel, message)
     val ask: Future[GraphMessage] = dbAskWritePropagate(rpcMessage)
-    Await.result(ask, duration)
+    Await.result(ask, duration) match {
+      case Left(msg) => {
+        val channelChirp: Channel = rpcMessage.getParamsChannel
+        channelChirp.decodeSubChannel match {
+          case(Some(lao_id)) => {
+            val broadcastChannel: Channel = Channel(Channel.rootChannelPrefix + Base64Data.encode(lao_id) + LaoHandler.SOCIALMEDIAPOSTSPREFIX)
+            rpcMessage.getParamsMessage match {
+              case Some(params) => {
+                // we can't get the message_id as a Base64Data, it is a Hash
+                val chirpId: Hash = params.message_id
+                val timestamp: Timestamp = params.decodedData.get.asInstanceOf[AddChirp].timestamp
+                val addBroadcastChirp: AddBroadcastChirp = AddBroadcastChirp(chirpId, channelChirp.channel, timestamp)
+                val broadcastData: Base64Data = Base64Data.encode(addBroadcastChirp.toJson.toString) //FIXME: check json/string conversion
```

Comment:
This is no longer a valid FIXME or is it ?

Reply:
Indeed this comment was forgotten here

Reply:
going to do this